Content.Server/Chemistry/ReagentEffects/RemoveComponent.cs

```diff
@@ -0,0 +1,64 @@
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.Localizations;
+using JetBrains.Annotations;
+using Robust.Shared.Prototypes;
+using System.Text;
+
+namespace Content.Server.Chemistry.ReagentEffects;
+
+/// <summary>
+/// Removes designated component or components.
+/// </summary>
+[UsedImplicitly]
+public sealed partial class RemoveComponent : ReagentEffect
+{
+    /// <summary>
+    /// ID of component to remove, as a string
+    /// Note: component ID shouldn't have the "component" postfix
+    /// </summary>
+    [DataField("components")]
+    public HashSet<string> Components = new();
+
+    protected override string? ReagentEffectGuidebookText(IPrototypeManager prototype, IEntitySystemManager entSys)
+    {
+        var guidebookComponents = new List<string>();
+
+        foreach (var entry in Components)
+        {
+            var entryText = entry.ToString(); // Not really sure if this is necessary but just to be safe
+            StringBuilder formattedText = new StringBuilder(entryText.Length * 2);
+            formattedText.Append(entryText[0]);
+            for (int i = 1;
+                i < entryText.Length;
+                i++)
+            {
+                if (char.IsUpper(entryText[i]) && entryText[i - 1] != ' ')
+                    formattedText.Append(' ');
+                formattedText.Append(entryText[i]);
+            }
+            guidebookComponents.Add(formattedText.ToString().ToLower() + 's');
+        }
```

Comment:
```suggestion
var guidebookComponents = new List<string>(Components.Count);

        foreach (var entry in Components)
        {
            var formattedText = new StringBuilder(entry.Length * 2);
            formattedText.Append(entry[0]);
            for (var i = 1;
                i < entry.Length;
                i++)
            {
                if (char.IsUpper(entry[i]) && entry[i - 1] != ' ')
                    formattedText.Append(' ');
                formattedText.Append(entry[i]);
            }
            guidebookComponents.Add(formattedText.ToString().ToLower() + 's');
        }
        }
```

The .ToString() is unnecessary.

Reply:
also bruh that for loop over 3 lines real